### ISSUE TITLE
Support one to many relationships

### ISSUE THREAD

**Issue by jdkandersson** [jdkandersson/OpenAlchemy#16]
As a user I want to define one to many relationships in the OpenAPI specification so that I can use them in my SQLAlchemy models.

**Comment by Shihta** [comment]
Does this mean to support array?

**Comment by jdkandersson** [comment]
Yes, it would, for example, allow for:
```yaml
Schemas:
  Employee:
    ...
  Division:
    ...
    properties:
      ...
      employees:
        type: array
        items:
          "$ref": "#/.../Employee"
```
Which would construct a one to many relationship between division and employee. Is that what you would expect to happen? Are you looking for any other kind of array support?

**Comment by Shihta** [comment]
yes! that's just what I want!
Is there any schedule on this feature?

**Comment by jdkandersson** [comment]
It should be shipped over the next week or two.

**Comment by jdkandersson** [comment]
To support one to many relationships, the child table must be modified upon discovering the relationship on the parent. This may require modifying the model after it has been created or it requires modifying the schema if it has not been created yet.

SQLAlchemy allows for defining further class attributes after the class definition. For example:
```python
class Division(models.Base):
    __tablename__ = "division"
    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)

setattr(Division, "name", sa.Column(sa.String))
```
is valid. The _setattr_ call can be made any time. If the models are relied on for table creation, the _setattr_ call must be made before the the table creation (eg. _db.create_all()_ in the case of _Flask-SQLAlchemy_).

Similar to the foreign key created for the many to one relationship, the foreign key that is created should not be included if the model instance is converted to a dictionary using _to_dict_. This means it must be marked in a special way in the schema associated with the child model to ensure it does not get recorded.

For the parent _to_dict_, the array schema must be recorded with the object item type. This means that it must be checked for and each item converted to a dictionary before returning.

For the parent _from_dict_, each child must be created before the parent and then passed in as an array to the parent construction.

**Comment by jdkandersson** [comment]
@Shihta included in release 0.9.0 available now.
